source/fundamentals/indexes.txt

```diff
@@ -0,0 +1,340 @@
+
+.. literalinclude:: /includes/fundamentals/code-snippets/IndexPage.java
+   :language: java
+   :dedent:
+   :start-after: begin covered compound query
+   :end-before: end covered compound query
+.. driver-content-end
+
+.. common-content-begin
+
+See the MongoDB server manual section on :manual:`Compound indexes </core/index-compound>` for more information.
+
+Multikey Indexes (Indexes on Array Fields)
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+**Multikey indexes** are indexes that improve performance for queries that specify a field with an index that contains
+an array value. You can define a multikey index using the same syntax as a single field or compound index.
+
+The following example creates a compound, multikey index on the ``rated``, ``genres`` (an array of Strings), and
+``title`` fields:
+
+.. common-content-end
+.. driver-content-begin
+.. literalinclude:: /includes/fundamentals/code-snippets/IndexPage.java
+   :language: java
+   :dedent:
+   :start-after: begin multikey index
+   :end-before: end multikey index
+
+The following is an example of a query that would be covered by the index created above:
+
+.. literalinclude:: /includes/fundamentals/code-snippets/IndexPage.java
+   :language: java
+   :dedent:
+   :start-after: begin covered multikey query
+   :end-before: end covered multikey query
+.. driver-content-end
+
+.. common-content-begin
+
+Multikey indexes behave differently from non-multikey indexes in terms of query coverage, index bound computation, and
+sort behavior. For a full explanation of multikey indexes, including a discussion of their behavior and limitations,
+refer to the :manual:`Multikey Indexes page </core/index-multikey>` in the MongoDB manual.
+
+Text Indexes
+~~~~~~~~~~~~
+
+**Text indexes** support text search queries on string content. These indexes can include any field whose value is a
+string or an array of string elements. MongoDB supports text search for various languages. You can specify the default
+language as an option when creating the index.
+
+.. warning::
+
+   Collections are limited to one text index. Keep in mind that a text index can cover multiple fields.
+
+.. common-content-end
+
+.. driver-content-begin
+
+The following example creates a text index on the ``plot`` field:
+
+.. literalinclude:: /includes/fundamentals/code-snippets/IndexPage.java
+   :language: java
+   :dedent:
+   :start-after: begin text index
+   :end-before: end text index
+
+The following is an example of a query that would use the index created above. Note that the ``sort`` is
+omitted because text indexes do not contain sort order.
+
+.. literalinclude:: /includes/fundamentals/code-snippets/IndexPage.java
+   :language: java
+   :dedent:
+   :start-after: begin text query
+   :end-before: end text query
+
+.. driver-content-end
+.. common-content-begin
+
+For a full explanation of text search with MongoDB, refer to :manual:`Text Indexes </core/index-text>` in the MongoDB
```

Comment:
Maybe worth mentioning Atlas Search here? Maybe controversial as it is Atlas only, but could be a better solution for users on Atlas than text indexes.